src/common/com/intellij/plugins/haxe/model/HaxeMemberModel.java

```diff
@@ -97,6 +97,7 @@ public static HaxeMemberModel fromPsi(PsiElement element) {
     if (element instanceof HaxeVarDeclaration) return new HaxeFieldModel((HaxeVarDeclaration)element);
     if (element instanceof HaxeLocalVarDeclaration) return new HaxeLocalVarModel((HaxeLocalVarDeclaration)element);
     if (element instanceof HaxeParameter) return new HaxeParameterModel((HaxeParameter)element);
+    if (element instanceof HaxeForStatement) return null;
```

Comment:
If I get this correctly (and I haven't tried it myself), we stop looking for variables at the for statement.  Does this limit the variables that are actually seen?  See the example below.